### PR TITLE
RFC: Collection of small fixes

### LINKED ISSUE #47: handle_no_nailgun cannot be in cement
The function handle_no_nailgun, which shall report import errors with nailgun is unavailable in case, nailgun cannot be imported, because cement uses nailgun components itself.
This must be bootstrapped another way.

### PR DESCRIPTION
**Pull request by mdellweg**
I am collecting small fixups on this branch i stumble across.
I do not want to make a Pull request for all of them, but eventually squash them thematically in no more than a handfull commits.
It is in no way meant to be a review covering the full codebase (to lazy).
Feel free to add more of those unrelated commits, or to cherry-pick things you think are urgent.

Effects:
https://github.com/theforeman/foreman-ansible-modules/issues/32
https://github.com/theforeman/foreman-ansible-modules/issues/29

fixes #47

### CODE REVIEW

**Comment by mdellweg** [comment]
@evgeni what do you think about this? Primarily i'd like to see the last one (Remove EntityCompareMixin) merged, to remove a sore spot on my git-blame profile.

**Comment by evgeni** [comment]
748bb48 - Fixes in foreman_global_parameter - ACK
2525d58 - Fixups in katello_upload - **NACK**, the doc string is wrong, it changes src back to file, and drops the alias we added, also src does not need no_log imho. also it does not support check_mode.
671b665 - Fixups in katello_activation_key - ACK
4767a71 - Fixups in katello_sync - ACK
bbaf86c - Fixes in katello_content_view - ACK
cea9a8b - Fixup katello_lifecycle_environment - ACK
b6ac57e - fixup katello_manifest - ACK
aeb6737 - Fixup katello_product - ACK
55bc279 - Fixup katello_publish - ACK
f69195e - Fixes in katello_repository - ACK
2604ecd - set verify_ssl default=True - ACK
03e1f39 - Remove EntityCompareMixin in favour of fields_equal - ACK

**Comment by mdellweg** [comment]
Updated. Fixed.
Thanks @evgeni, i think this happened because it were a lot of repetitive tasks.

File: modules/katello_upload.py [hunk 1]
```diff
@@ -141,13 +142,13 @@ def main():
             server_url=dict(required=True),
             username=dict(required=True, no_log=True),
             password=dict(required=True, no_log=True),
-            verify_ssl=dict(required=False, type='bool', default=False),
-            src=dict(required=True, no_log=True, aliases=['file']),
-            repository=dict(required=True, no_log=False),
-            product=dict(required=True, no_log=False),
-            organization=dict(required=True, no_log=False),
+            verify_ssl=dict(type='bool', default=True),
+            src=dict(required=True, aliases['file']),
```
Comment (evgeni) [inline modules/katello_upload.py]: missing `=`

Comment (mdellweg) [reply modules/katello_upload.py]: sorry, you can have mine... `=`

**Comment by mdellweg** [comment]
Also rebased

File: modules/foreman_global_parameter.py [hunk 2]
```diff
@@ -149,11 +150,8 @@ def main():
         ),
         supports_check_mode=True,
     )
-    if not HAS_NAILGUN_PACKAGE:
-        module.fail_json(
-            msg="""Missing required nailgun module (check docs or install)
-            with: pip install nailgun"""
-        )
+
+    handle_no_nailgun(module, HAS_NAILGUN_PACKAGE)
```
[the remark this answers is not in the document]
Comment (mdellweg) [reply modules/foreman_global_parameter.py]: How shall we handle this in general?

[the remark this answers is not in the document]
Comment (akofink) [reply modules/foreman_global_parameter.py]: Let's discuss on the issue for ideas.

**Comment by mdellweg** [comment]
@akofink, @evgeni rebased

File: modules/katello_content_view.py [hunk 3]
```diff
@@ -80,6 +81,9 @@
 except:
     HAS_NAILGUN_PACKAGE = False
 
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.foreman_helpers import handle_no_nailgun
```
Comment (akofink) [inline modules/katello_content_view.py]: s/foreman_helpers/foreman_helper/

**Comment by akofink** [comment]
Just one typo; everything else looks good.

File: modules/katello_content_view.py (hunk 3 above)
Comment (mdellweg) [reply modules/katello_content_view.py]: You really seem to have a testing environment for all that stuff...
And i get lazy after doing the same change in about 6 different files.

Comment (akofink) [reply modules/katello_content_view.py]: That one's just vim and the_silver_searcher :wink:

Comment (akofink) [reply modules/katello_content_view.py]: But, we should *definitely* make a test suite.

**Pull request merged**